sdk/synapse/synapse-spark/review/synapse-spark.api.md

```diff
@@ -0,0 +1,525 @@
+    executorMemory?: string;
+    // (undocumented)
+    file: string;
+    // (undocumented)
+    files?: string[];
+    // (undocumented)
+    jars?: string[];
+    // (undocumented)
+    name: string;
+    // (undocumented)
+    pythonFiles?: string[];
+    tags?: {
+        [propertyName: string]: string;
+    };
+}
+
+// @public
+export type SparkBatchJobResultType = "Uncertain" | "Succeeded" | "Failed" | "Cancelled" | string;
+
+// @public (undocumented)
+export interface SparkBatchJobState {
+    currentState?: string;
+    deadAt?: Date;
+    // (undocumented)
+    jobCreationRequest?: SparkRequest;
+    notStartedAt?: Date;
+    recoveringAt?: Date;
+    runningAt?: Date;
+    startingAt?: Date;
+    successAt?: Date;
+    terminatedAt?: Date;
+}
+
+// @public (undocumented)
+export class SparkClient {
+    constructor(workspaceEndpoint: string, sparkPoolName: string, credential: TokenCredential, pipelineOptions?: SparkClientOptions);
+    cancelSparkBatchJob(batchId: number, options?: CancelSparkBatchJobOptions): Promise<OperationResponse>;
+    cancelSparkSession(sessionId: number, options?: CancelSparkSessionOptions): Promise<OperationResponse>;
+    cancelSparkStatement(sessionId: number, statementId: number, options?: CancelSparkStatementOptions): Promise<OperationResponse>;
+    createSparkBatchJob(sparkBatchJobOptions: SparkBatchJobOptions, options?: CreateSparkBatchJobOptions): Promise<CreateSparkBatchJobResponse>;
+    createSparkSeesion(sparkSessionOptions: SparkSessionOptions, options?: CreateSparkSessionOptions): Promise<CreateSparkSessionResponse>;
+    createSparkStatement(sessionId: number, sparkStatementOptions: SparkStatementOptions, options?: CreateSparkStatementOptions): Promise<CreateSparkStatementResponse>;
+    getSparkBatchJob(batchId: number, options?: GetSparkBatchJobOptions): Promise<GetSparkBatchJobResponse>;
+    getSparkSession(sessionId: number, options?: GetSparkSessionOptions): Promise<GetSparkSessionResponse>;
+    getSparkStatement(sessionId: number, statementId: number, options?: GetSparkStatementOptions): Promise<GetSparkStatementResponse>;
+    listSparkBatchJobs(options?: ListSparkBatchJobsOptions): Promise<ListSparkBatchJobsResponse>;
+    listSparkSessions(options?: ListSparkSessionsOptions): Promise<ListSparkSessionsResponse>;
+    listSparkStatements(sessionId: number, options?: ListSparkStatementsOptions): Promise<ListSparkStatementsResponse>;
+    resetSparkSessionTimeout(sessionId: number, options: ResetSparkSessionTimeoutOptions): Promise<OperationResponse>;
+    readonly workspaceEndpoint: string;
+}
+
+// @public
+export interface SparkClientOptions extends PipelineOptions {
+    endpoint?: string;
+    livyApiVersion?: string;
+}
+
+// @public
+export type SparkErrorSource = "System" | "User" | "Unknown" | "Dependency" | string;
+
+// @public
+export type SparkJobType = "SparkBatch" | "SparkSession" | string;
+
+// @public (undocumented)
+export interface SparkRequest {
```

Comment:
this seems to share a lot of surface with `SparkBatchJobOptions` and `SparkBatchJob`  - can we reuse types somehow?

Reply:
We plan to fix it when board review, that is a common issue between all languages.

---
In reply to: [521610642](https://github.com/Azure/azure-sdk-for-js/pull/12487#discussion_r521610642) [](ancestors = 521610642)